Example/Unit Tests/CGAffineTransformDecompositionTests.swift

```diff
@@ -0,0 +1,311 @@
+        ) {
+            assertEqual(
+                CGAffineTransform.identity
+                    .rotated(by: angle)
+                    .decomposed(),
+                CGAffineTransform.DecomposedMatrix(
+                    rotation: decomposedAngle ?? angle
+                ),
+                accuracy: 1e-10,
+                file: file,
+                line: line
+            )
+        }
+
+        assertDecomposesRotation(0)
+        assertDecomposesRotation(.pi / 4)
+        assertDecomposesRotation(.pi / 2)
+        assertDecomposesRotation(.pi)
+        assertDecomposesRotation(-.pi / 4)
+        assertDecomposesRotation(-.pi / 2)
+        assertDecomposesRotation(-.pi)
+
+        // Rotating past 180º is the same as rotating a smaller angle in the opposite direction. The decomposed angle
+        // should be the smallest angle resulting in the same transition.
+        assertDecomposesRotation(.pi * 1.5, decomposedAngle: -.pi / 2)
+        assertDecomposesRotation(-.pi * 1.5, decomposedAngle: .pi / 2)
+
+        // Rotating past a full rotation is equivalent to removing some number of full rotations and rotating the
+        // remainder of the way. The decomposed angle should be the smallest angle resulting in the same transform.
+        assertDecomposesRotation(2 * .pi, decomposedAngle: 0)
+        assertDecomposesRotation(3 * .pi, decomposedAngle: .pi)
+        assertDecomposesRotation(4 * .pi, decomposedAngle: 0)
+        assertDecomposesRotation(-2 * .pi, decomposedAngle: 0)
+        assertDecomposesRotation(-3 * .pi, decomposedAngle: -.pi)
+    }
+
+    func testDecompositionRotationAndScale() {
+        assertEqual(
+            CGAffineTransform.identity
+                .rotated(by: .pi / 4)
+                .scaledBy(x: 3, y: 2)
+                .decomposed(),
+            CGAffineTransform.DecomposedMatrix(
+                scaleX: 3,
+                scaleY: 2,
+                rotation: .pi / 4
+            ),
+            accuracy: 1e-10
+        )
+    }
+
+    // MARK: - Tests - Recomposition
+
+    func testRecompositionIdentity() {
+        XCTAssertEqual(CGAffineTransform.DecomposedMatrix().recompose(), CGAffineTransform.identity)
+    }
+
+    func testRecompositionTranslation() {
+        func assertRecomposesTranslation(x: CGFloat, y: CGFloat, file: StaticString = #file, line: UInt = #line) {
+            XCTAssertEqual(
+                CGAffineTransform.DecomposedMatrix(
+                    translateX: x,
+                    translateY: y
+                ).recompose(),
+                CGAffineTransform.identity
+                    .translatedBy(x: x, y: y),
+                file: file,
+                line: line
+            )
+        }
+
+        assertRecomposesTranslation(x: 10, y: 20)
+        assertRecomposesTranslation(x: -4, y: -5)
+        assertRecomposesTranslation(x: 10, y: 0)
+        assertRecomposesTranslation(x: 0, y: 10)
+    }
+
+    func testRecompositionScale() {
```

Comment:
This test brought up an interesting problem: a scale of `x: -3, y: -2` produces an identical transform as a scale of `x: 3, y: 2` combined with a rotation of 180º.

As a future optimization, I think we could return an array of possible decompositions, and determine the closest pair between which we can interpolate. For example, interpolating from that transform to a scale of `x: 3, y: 2` would use the negative scale decomposition and interpolate the scale only; whereas interpolating from that transform to a scale of `x: 3, y: 2` with a rotation of 90º would treat it as a positive scale with a rotation, so we would only interpolate the rotation and not the scale.

That definitely increases the complexity, but I think it would produce some nice results.